packages/css/src/getFileAndDirFromPath.ts

```diff
@@ -0,0 +1,69 @@
+const getLastSlashBeforeIndex = (path: string, index: number) => {
+  let pathIndex = index - 1;
+
+  while (pathIndex >= 0) {
+    if (path[pathIndex] === '/') {
+      return pathIndex;
+    }
+
+    pathIndex--;
+  }
+
+  return -1;
+};
+
+type FileAndDir = {
+  file: string;
+  dir?: string;
+};
+
+/**
+ * Assumptions:
+ * - The path is always normalized to use posix file separators (/) (see `addFileScope`)
+ * - The path is always relative to the project root, i.e. there will never be a leading slash (see `addFileScope`)
+ * - As long as `.css` is there, we have a valid `.css.*` file path, because otherwise there wouldn't
+ *   be a file scope to begin with
+ */
+const _getFileAndDirFromPath = (path: string): FileAndDir | undefined => {
+  let file: string;
+  let dir: string | undefined;
+
+  const lastIndexOfDotCss = path.lastIndexOf('.css');
+
+  if (lastIndexOfDotCss === -1) {
+    return;
+  }
+
+  const lastSlashIndex = getLastSlashBeforeIndex(path, lastIndexOfDotCss);
+  if (lastSlashIndex === -1) {
+    return { file: path.slice(0, lastIndexOfDotCss) };
+  }
+
+  let secondLastSlashIndex = getLastSlashBeforeIndex(path, lastSlashIndex - 1);
+
+  // If secondLastSlashIndex is -1, it means that the path looks like `directory/file.css.ts`,
+  // in which case dir will still be sliced starting at 0, which is what we want
+  dir = path.slice(secondLastSlashIndex + 1, lastSlashIndex);
+  file = path.slice(lastSlashIndex + 1, lastIndexOfDotCss);
+
+  return { dir, file };
+};
+
+const memoizedGetFileAndDirFromPath = () => {
```

Comment:
Do you see a lot of hits on the cache? Just wondering if this is worth keeping as it'll leak memory slowly in a dev server right?

You could also consider using an LRU? 
https://www.npmjs.com/package/lru-cache

Reply:
Every identifier generated in a file except the first one will hit the cache, so there's a lot of hits, especially in something like a sprinkles file. I think it would take quite a large app to have a meaningful memory leak, but it's still probably a good idea to use an lru cache.